tests/tensorlake/test_routing.py

```diff
@@ -0,0 +1,94 @@
+import unittest
+from typing import List, Union
+
+import parameterized
+from testing import remote_or_local_graph, test_graph_name
+
+from tensorlake import Graph, RouteTo, tensorlake_function
+
+
+@tensorlake_function()
+def a(value: int) -> int:
+    return value + 3
+
+
+@tensorlake_function()
+def b(value: int) -> int:
+    return value + 4
+
+
+@tensorlake_function()
+def c(value: int) -> int:
+    return value + 5
+
+
+@tensorlake_function(next=[a, b])
+def fan_out(value: int) -> int:
+    return value
+
+
+@tensorlake_function(next=[a, b, c])
+def route_out(value: int) -> RouteTo[int, Union[a, b, c]]:
+    if value % 2 == 0:
+        return RouteTo(value + 1, [a])
+
+    return RouteTo(value + 2, [b, c])
+
+
+@tensorlake_function(accumulate=int)
+def sum_of_squares(current: int, value: int) -> int:
+    return current + value
+
+
+@tensorlake_function(next=sum_of_squares)
+def square_values(value: int) -> int:
+    return value * value
+
+
+@tensorlake_function(next=square_values)
+def parallel_map(count: int) -> List[int]:
+    return list(range(count))
+
+
+_GRAPH_PARAM_SETS = [(False), (True)]
```

Comment:
fyi parameterized.expand supports naming of test cases (and possibly parameter values). I mean `_GRAPH_PARAM_SETS = [(False), (True)]` is very implicit when used a global variable away from the actual test definition that has `is_remote` parameter names.

Reply:
Done :-)